src/fetch/transfer_file.py

```diff
@@ -362,31 +364,87 @@ def __str__(self) -> str:
         )
 
 
+def map_reward(
+    amount: float,
+    risk_free: bool,
+    batch_contains_unsafe_liquidity: bool,
+) -> float:
+    """
+    Converts orderbook rewards based on additional knowledge of "risk_free" transactions
+    """
+    if amount > 0 and risk_free and not batch_contains_unsafe_liquidity:
+        # Risk Free Orders that are not liquidity orders get 37 COW tokens.
+        return 37.0
+    return amount
+
+
+def liquidity_order_batches(order_df: DataFrame) -> set[str]:
+    """
+    Fetches the set of transaction hashes containing a
+    liquidity order as the unique tx_hash where reward `amount = 0`
+    """
+    order_df = order_df.astype({"safe_liquidity": "boolean"})
+    liquidity = order_df.loc[order_df["amount"] == 0]
+    # Pandas doesn't seem to work with boolean types:
+    # when using "is False" we get
+    # KeyError: 'False: boolean label can not be used without a boolean index'
+    # When using == False we get pylint error
+    unsafe_liquidity = liquidity.loc[
+        liquidity["safe_liquidity"] == False  # pylint: disable=singleton-comparison
+    ]
+    return set(unsafe_liquidity["tx_hash"].unique())
```

Comment:
Do you need the unique, if you anyways create a set form it?

Reply:
Perhaps not. However, the set here was intended to be a casting operation because pandas.unique returns a list for some silly reason. I could remove the unique here if you prefer.

Reply:
I will also add a unit test for this method "unsafe_batches" as well - to confirm that it does what we want (with or without unique).